File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    citations.sort((a,b)=>a-b);
    
    const map = new Map();
    citations.map(v => {
        if (map.has(v)) map.set(v, map.get(v)+1);
        else map.set(v, 1);
    });
    // console.log(map)
    
    let len = citations.length;
    let maxh = 0;
    for (let i=0; i<citations[citations.length-1]; i++) {
        if (len>=i) {
            // console.log('실행');
            if (map.has(i)) len-=map.get(i);
            maxh = i;
            // console.log('maxh', maxh);
        } else break;
    }
    return maxh;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort()
    start=0
    end=citations[-1]
    answer = 0
    while start<=end:
        mid = (start+end)//2
        idx = -1
        for i in range(len(citations)):
            if citations[i]>=mid:
                idx = i
                break
        if len(citations)-idx>=mid:
            answer = max(answer,mid)
            start = mid+1
        else:
            end = mid-1
    return answer